### PR TITLE
[PoC] Futures in place of actors

### PR DESCRIPTION
**Pull request by wpopielarski**
Please look at ScalaJdiEventDispatcher, its test and ScalaDebugTarget.
It shows a direction I'm heading in debugger refactoring. Are you OK
with it?
ps. I know there is a little mess on this branch. Goin' to unwind it.

### CODE REVIEW

**Comment by sschaef** [comment]
Hey @wpopielarski, your commits look good, not sure if you planned to squash them. Don't because if you improve the commit messages a little bit, they are fine in my opinion. Also, you can rebase on master, the first commit is already merged into master.

**Comment by wpopielarski** [comment]
retest this please

**Comment by sschaef** [comment]
I think this branch can now be rebased on master. Also, the first commit needs to be skipped and the second commit could get a clean up of its commit message.

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtInputs.scala [hunk 1]
```diff
@@ -36,7 +36,8 @@ import org.scalaide.core.internal.ScalaPlugin
 import org.eclipse.core.resources.IContainer
 import org.eclipse.core.runtime.IPath
 
-/** Inputs-like class, but not implementing xsbti.compile.Inputs.
+/**
+ * Inputs-like class, but not implementing xsbti.compile.Inputs.
```
Comment (sschaef) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/zinc/SbtInputs.scala]: Please review your formatting options. These partly doc changes shouldn't happen. Either let them as they are or format them correctly, i.e. one space indent for every line, starting on the second line.

File: org.scala-ide.sdt.debug.tests/src/org/scalaide/debug/internal/hcr/HotCodeReplaceTest.scala [hunk 2]
```diff
@@ -22,8 +23,26 @@ import org.scalaide.core.testsetup.SDTTestUtils
...
+import HotCodeReplaceTest.MainMethodBeginning
+import HotCodeReplaceTest.MainMethodEnd
+import HotCodeReplaceTest.Matcher
+import HotCodeReplaceTest.RecursiveMethodBeginning
+import HotCodeReplaceTest.RecursiveMethodEnd
+import HotCodeReplaceTest.RecursiveMethodSelfCall
+import HotCodeReplaceTest.RunMethodBeginning
+import HotCodeReplaceTest.RunMethodEnd
+import HotCodeReplaceTest.TestHcrSuccessListener
+import HotCodeReplaceTest.TestedFilePath
```
Comment (sschaef) [inline org.scala-ide.sdt.debug.tests/src/org/scalaide/debug/internal/hcr/HotCodeReplaceTest.scala]: These imports are unnecessary.

File: org.scala-ide.sdt.debug.tests/src/org/scalaide/debug/internal/model/DebugTargetTerminationTest.scala [hunk 3]
```diff
@@ -33,83 +31,34 @@ import com.sun.jdi.VirtualMachine
...
-        }
-      }
-    }
-  }
-  /**A dummy runtime exception for testing purposes. You should use throw this exception only when testing code that is expected
-   * to gracefully recover from generic exceptions.*/
+  /**
+   * A dummy runtime exception for testing purposes. You should use throw this exception only when testing code that is expected
+   * to gracefully recover from generic exceptions.
+   */
```
Comment (sschaef) [inline org.scala-ide.sdt.debug.tests/src/org/scalaide/debug/internal/model/DebugTargetTerminationTest.scala]: This is a correct comment change. ;)

File: org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/JdiEventHandling.scala [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package org.scalaide.debug.internal
+
+import com.sun.jdi.event.Event
+import scala.concurrent.Future
+import com.sun.jdi.request.EventRequest
+import scala.concurrent.ExecutionContext
+
+trait JdiEventDispatcher {
```
Comment (sschaef) [inline org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/JdiEventHandling.scala]: Scaladoc for these types is needed.

File: org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/breakpoints/BreakpointSupport.scala [hunk 5]
```diff
@@ -31,30 +35,27 @@ import com.sun.jdi.event.Event
...
+import BreakpointSupportSubordinate.createBreakpointRequest
+import BreakpointSupportSubordinate.handleEvent
 import RichBreakpoint.richBreakpoint
 
 private[debug] object BreakpointSupport {
   /** Attribute Type Name */
   final val ATTR_TYPE_NAME = "org.eclipse.jdt.debug.core.typeName"
 
-  /** Create the breakpoint support actor.
+  /**
```
Comment (sschaef) [inline org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/breakpoints/BreakpointSupport.scala]: This doc comment is off. Also it needs to be corrected.

**Comment by sschaef** [comment]
I cherry picked the last commit message, you can actually remove that commit once the other PR is merged and you rebased to master.

**Comment by sschaef** [comment]
I removed the mentioned issues of this PR in #993, we should continue to work on that branch.

**Comment by sschaef** [comment]
Superseded by #993

**Pull request closed without merging**